### PR TITLE
Separates MPI and non-MPI configurations for scons

### PR DESCRIPTION
**Pull request by rarutter**
This should improve robustness for building on different
platforms and with different compilers by not including anything
MPI related without it being manually specified.

### CODE REVIEW

**Comment by rarutter** [comment]
This involves issues #221 and #68 but may not completely address them.

File: SConstruct [hunk 1]
```diff
@@ -149,8 +145,10 @@ if comp_name == 'aeshna': #aeshna... check name
 
 #atlas?
```
Comment (tobeycarman) [inline SConstruct]: In order to get this to work on atlas, I had to add the following code here:

```
if comp_name == 'atlas.snap.uaf.edu':
  platform_libs[:] = [lib for lib in platform_libs if not lib == 'jsoncpp']
  platform_libs.append('json_linux-gcc-4.4.7_libmt')
  platform_include_path.append('/home/UA/tcarman2/boost_1_55_0/')
  platform_include_path.append('/home/UA/rarutter/include')

  platform_library_path.append('/home/UA/rarutter/lib')
  platform_library_path.append('/home/UA/tcarman2/boost_1_55_0/stage/lib')
```

As well as hand-manipulating the order of the `-L` flags for the load step of compiling. Not sure yet the best way to address this directly in the SConstruct file.

File: SConstruct [hunk 2]
```diff
@@ -118,7 +114,7 @@ if platform_name == 'Linux': #rar, tobey VM, Colin, Vijay, Helene VM(?)
   platform_libs = libs
 
 
-elif platform_name == 'Darwin': #tobey
+elif platform_name == 'Darwin':
```
Comment (tobeycarman) [inline SConstruct]: In order to get this to work on my OS X machine, I had to add the following code here:

```
  # On OSX, using Homebrew, alternate g++ versions are installed so as not
  # to interfere with the system g++, so here, we have to set the compiler
  # to the specific version of g++ that we need.
  compiler = distutils.spawn.find_executable('g++-4.8')
```

**Comment by tobeycarman** [comment]
@rarutter, I feel like this is ready to merge. I don't think it will necessarily fix all issues for compiling on other computers, but it is at least an improvement. Actually being able to use Scons on Atlas will be really helpful.

**Pull request merged**